packages/frontend/src/utils/getTranslationsFromMultisigRequest.js

```diff
@@ -0,0 +1,93 @@
+import BN from 'bn.js';
+import { utils } from 'near-api-js';
+
+const {
+    format: {
+        formatNearAmount
+    },
+    key_pair: {
+        PublicKey,
+    },
+} = utils;
+
+
+const formatNear = (amount) => {
+    try {
+        return formatNearAmount(amount, 4);
+    } catch {
+        // Near amount may be stored as hex rather than decimal
+        return formatNearAmount(new BN(amount, 16).toString(), 4);
+    }
+};
+
+const rawPublicKeyToString = ({ keyType, data }) => new PublicKey({ keyType, data }).toString();
+
+const parseAndFormatArguments = (argsBuffer) => {
+    const args = JSON.parse(Buffer.from(argsBuffer).toString());
+
+    const parsedArgs = {
+        ...args,
+        ...(args.amount && { amount: formatNear(args.amount) }),
+        ...(args.deposit && { deposit: formatNear(args.deposit )}),
+    };
+
+    return JSON.stringify(parsedArgs, null, 2);
+};
+
+export default function getTranslationsFromMultisigRequest({ actions, receiverId, accountId }) {
+    const fullAccessKeyAction = actions.find(({ enum: type, permission }) => type === 'addKey' && !permission);
```

Comment:
I'm getting full access key warning modal when recovering my account secured with email link + 2FA despite no full access key being added. The 2FA email that I received from [contract-helper](https://github.com/near/near-contract-helper/blob/436762c54a519b400692c17d2ead910a1a589fdf/middleware/2fa.js#L143) is showing the correct information though.

Looks like it's `accessKey.permission`

<img width="563" alt="Screen Shot 2022-03-16 at 12 50 42 PM" src="https://user-images.githubusercontent.com/24921205/158677884-130e98fd-acf4-43c3-b9a7-c87f1d343c31.png">

Reply:
Good catch,  I wasn't even parsing out the action itself either. Fixed 👍🏽